### PR TITLE
migrate to angular 18

### PR DESCRIPTION
**Pull request by ajaygawali120892**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,7 +32,7 @@ npm install @vinlos/ngx-gallery --save
 
 ### Versions
 
-16.x.x and above - for Angular v16+
+18.x.x and above - for Angular v18+
```
Comment (vinlos) [inline README.md]: I think we can state that old versions will continue to work with older Angular versions. So, 16+ is still valid

**Review by vinlos — CHANGES_REQUESTED** [review]
we also need to update peer dependencies in library package.json

File: package.json [hunk 2]
```diff
@@ -20,48 +20,47 @@
...
-    "@types/node": "^20.12.4",
-    "@typescript-eslint/eslint-plugin": "7.5.0",
-    "@typescript-eslint/parser": "7.5.0",
+    "@types/node": "^22.7.9",
+    "@typescript-eslint/eslint-plugin": "^8.11.0",
     "coveralls": "^3.1.1",
-    "cpx": "^1.5.0",
-    "eslint": "^8.57.0",
-    "jasmine-core": "~5.1.2",
+    "cpx": "^1.2.1",
```
Comment (vinlos) [inline package.json]: I don't think we need cpx and coveralls

**Comment by vinlos** [comment]
> we also need to update peer dependencies in library package.json

this isn't done yet

**Comment by ajaygawali120892** [comment]
> > we also need to update peer dependencies in library package.json
> 
> this isn't done yet

I run `npm outdated` now it shows nothing. Is there specific command or step to update peer dependancy ?

File: projects/gallery/package.json [hunk 3]
```diff
@@ -3,20 +3,22 @@
...
   "bugs": {
     "url": "https://github.com/vinlos/ngx-gallery/issues"
   },
   "peerDependencies": {
-    "@angular/common": ">=17.3.2",
-    "@angular/core": ">=17.3.2",
-    "@angular/animations": ">=17.3.2"
+    "@angular/common": "^18.2.9",
+    "@angular/core": "^18.2.9",
+    "@angular/animations": "^18.2.9"
```
Comment (vinlos) [inline projects/gallery/package.json]: for peer dependencies, we don't want to be so restrictive. We can then just use `>=`

Comment (ajaygawali120892) [reply projects/gallery/package.json]: updated

**Review by vinlos — APPROVED** [review]

**Pull request merged**